src/caikit_nlp_client/grpc_client_introspection.py

```diff
@@ -10,14 +12,61 @@
 log = logging.getLogger(__name__)
 
 
-class GrpcCaikitNlpClientIntrospection:
+@dataclass
+class GrpcConfig:
+    host: str
+    port: int
+    insecure: bool = False
+    ca_cert: Optional[bytes] = None
+    client_key: Optional[bytes] = None
+    client_cert: Optional[bytes] = None
+
+
+def make_channel(config: GrpcConfig) -> grpc.Channel:
+    log.debug(f"Making a channel from this config {config}")
+    if config.host.strip() == "":
+        raise ValueError("A non empty host name is required")
+    if config.port <= 0:
+        raise ValueError("A non zero port is required")
+
+    connection = f"{config.host}:{config.port}"
+
+    if config.insecure:
+        return grpc.insecure_channel(connection)
+
+    if config.client_key is None:
+        raise ValueError("A client key is required")
+    if config.client_cert is None:
+        raise ValueError("A client cert is required")
+    if config.ca_cert is None:
+        raise ValueError("A certificate authority certificate is required")
+
+    return grpc.secure_channel(
+        connection,
+        grpc.ssl_channel_credentials(
+            config.ca_cert, config.client_key, config.client_cert
+        ),
+    )
+
+
+class GrpcClient:
     """GRPC client for a caikit nlp runtime server
 
     Args:
         channel (grpc.Channel): a connected GRPC channel for use of making the calls.
     """
 
-    def __init__(self, channel: grpc.Channel) -> None:
+    def __init__(self, config: GrpcConfig) -> None:
```

Comment:
Changed the API so that it takes a config instead of a channel: the user will not care about the underlying channel

Reply:
channels need to be closed, having them exposed puts the onus of the client to manage the channels.